plugins/ipnet/resync_events.go

```diff
@@ -82,11 +82,12 @@ func (n *IPNet) Resync(event controller.Event, kubeStateData controller.KubeStat
 //  - inter-VRF routing
 //  - IP neighbor scanning
 func (n *IPNet) configureVswitchConnectivity(event controller.Event, txn controller.ResyncOperations) error {
+	// explicitly create Main/Pod VRF tables for IPv4/IPv6 (plugins and code depend on them)
 	if !n.test {
-		// explicitly create POD VRF using binary API
-		// TODO: just temporary, until VRF will be a separate NB API entity of vpp-agent,
-		// so that all plugins can properly set their dependency on it
-		n.createVrf(n.ContivConf.GetRoutingConfig().PodVRFID)
+		n.configureVRFTable(n.ContivConf.GetRoutingConfig().MainVRFID, vpp_l3.VrfTable_IPV4, "mainVRF", txn)
+		n.configureVRFTable(n.ContivConf.GetRoutingConfig().PodVRFID, vpp_l3.VrfTable_IPV4, "podVRF", txn)
+		n.configureVRFTable(n.ContivConf.GetRoutingConfig().MainVRFID, vpp_l3.VrfTable_IPV6, "mainVRF", txn)
+		n.configureVRFTable(n.ContivConf.GetRoutingConfig().PodVRFID, vpp_l3.VrfTable_IPV6, "podVRF", txn)
```

Comment:
Just for the sake readability we prefer to put all the configuration details to methods in pod/node/host.go (there is already a section for VRFs in node.go). Here we just call those methods as needed and put the returned configuration into the transaction.
Furthermore, you don't have to restrict this to non-test scenarios (if !n.test) - it was here before only because we would directly call the bin API, which is not handled in unit tests.
Consider this for example: https://github.com/milanlenco/vpp-1/commit/e8f2862a7981621fed7494b5f584c6e4d040011c

Reply:
redesigned in https://github.com/contiv/vpp/pull/1523/commits/cb15afb6ff614af58d97ed6003593e1cd8600f9e